lib/avro_patches/logical_types/decode_date_as_time_patch.rb

```diff
@@ -0,0 +1,14 @@
+# The avro-patches gem will convert some date-related logical-types in
+# Avro schemas to Ruby Date and Time objects.
+#
+# LogStash has methods to convert certain Ruby classes into JSON
+# values, but it does not support Ruby's Date class. It does support
+# Time (which is used by the timestamp-micros and timestamp-millis
+# logical types), so we convert the Date object to Time here.
+Avro::LogicalTypes::IntDate.class_eval do
+  EPOCH_START = Date.new(1970, 1, 1)
```

Comment:
This is not correct value of Epoch. It will translate to local timezone's 01 Jan 1970. We need to get UTC's 01 Jan 1970.